src/main/java/spiffe/api/svid/SpiffeEndpointChannelBuilder.java

```diff
@@ -0,0 +1,117 @@
+package spiffe.api.svid;
+
+import io.grpc.*;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import io.netty.channel.epoll.EpollDomainSocketChannel;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.kqueue.KQueueEventLoopGroup;
+import io.netty.channel.kqueue.KQueueServerDomainSocketChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.unix.DomainSocketAddress;
+import org.apache.commons.lang3.SystemUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+/**
+ *  Encapsulates the logic for creating the ManagedBuilder
+ *  Resolves the Address from Env if it's not provided as parameter.
+ *  Based on the Address scheme, creates a TCP or a Native Socket Channel
+ *  Configures the Socket Channel based on the detected OS
+ *
+ */
+class SpiffeEndpointChannelBuilder {
+
+    static final String ENV_ADDRESS_VAR = "SPIFFE_ENDPOINT_SOCKET";
+
+    /**
+     * Returns a configured ManagedChannel
+     * @param spiffeEndpointAddress
+     * @return
+     */
+    static ManagedChannel newChannel(String spiffeEndpointAddress) {
+        if (isBlank(spiffeEndpointAddress)) {
+            spiffeEndpointAddress = getAddressFromEnv();
+        }
+
+        URI parsedAddress = parseAddress(spiffeEndpointAddress);
+        if (isTcp(parsedAddress)) {
+            return createTcpChannel(parsedAddress);
+        } else {
+            return createNativeSocketChannel(spiffeEndpointAddress);
+        }
+    }
+
+    /**
+     * Try to resolve the Address from the Environment.
+     * @throws IllegalStateException if the Address is not found
+     * @return the Address
+     */
+    private static String getAddressFromEnv() {
+        String address = System.getenv(ENV_ADDRESS_VAR);
```

Comment:
Should we also consider using system properties? I think most java folks are used to use them instead of environment variables.

Reply:
The specification states: "Clients may be explicitly configured with the socket location, or may utilize the well-known environment variable SPIFFE_ENDPOINT_SOCKET. If not explicitly configured, conforming clients MUST fall back to the environment variable."

https://github.com/evan2645/spiffe/blob/3bdb4bd862e3abbb4167b4b75b8e0ecab48a3e3a/standards/SPIFFE_Workload_Endpoint.md#4-locating-the-endpoint